### DISCUSSION TITLE
Welcome to handlebars-helpers Discussions!

### DISCUSSION THREAD

**Discussion by jonschlinkert** [helpers/handlebars-helpers#389] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Fitzbert-Fitz** [comment]
Hi there, 
I am new to Handlebars and I want to use them inside CI/CD release notes context in azure devops. I followed these [instructions ](https://blogs.blackmarble.co.uk/rfennell/how-can-i-automatically-create-azure-devops-release-notes-and-how-can-i-publish-them/).

I am reading the description of a workitem which contains  a section `<<<ReleaseNotes>>> Our release notes<<<EndReleaseNotes>>>`.
`{{lookup this.fields 'System.Description'}}`  provides me the html description and `{{{lookup this.fields 'System.Description'}}}` provides me the raw content. 

So far so good. Now I want to find the mentioned section. I tried different stuff like e.g.
`{{match (lookup this.fields 'System.Description') ''<<<ReleaseNotes>>>\s(?s)(.*)<<<EndReleaseNotes>>>'' }}`

But it feels like whatever I do, I cannot get it working. Feels like I have some basic misunderstanding of using subexpressions or html escaping. Can someone give me a hint?

What do I need to do, that `{{match}}` returns "Our release notes".

Appreciate all hints!
Cheers Martin

**Reply by doowb** [reply]
`{{match}}` uses [micromatch](https://github.com/micromatch/micromatch) to match file paths using a glob pattern.

The only regex type helpers are `{{toRegex}}` and `{{test}}`. Depending on your case, you could use some of the array helpers to achieve what you need, or you could create another helper that does the regex matching.

Here's an idea using some of the array helpers:

```hbs
{{#if (test (lookup this.fields "System.Description") toRegex("<<<ReleaseNotes>>>\s(?s)(.*)<<<EndReleaseNotes>>>"))}}
  {{itemAt (split (itemAt (split (lookup this.fields "System.Description") "<<<ReleaseNotes>>>") -1) "<<<EndReleaseNotes>>>") 0}}
{{/if}}
```

1. The `test` helper is used to ensure the pattern exists and can be split.
2. the crazy amount of subexpressions works "inside-out" to get the string in the middle using the following steps:
    1. Split the original string on `<<<ReleaseNotes>>>`, which gives the following array:
        `['<<<ReleaseNotes>>>', ' Our release notes<<<EndReleaseNotes>>>']`
    2. Get the last item of the array:
        ` Our release notes<<<EndReleaseNotes>>>`
    3. Split the last item of the array on `<<<EndReleaseNotes>>>`, which gives the following array:
        `[' Our release notes', '<<<EndReleaseNotes>>>']`
    4. Get the first item of the array:
        ` Our release notes`

The first `itemAt` helper is using `-1` to get the last item of the array instead of getting the second item `1` because the result of `split` could include an empty string as the first item in the array.
